ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/fillers/DiscretePstTapFiller.java

```diff
@@ -93,17 +96,16 @@ private void buildPstTapVariablesAndConstraints(LinearProblem linearProblem, Pst
             RangeAction<?> preventiveRangeAction = lastAvailableRangeAction.getKey();
             Pair<Double, Double> pstLimits = getMinAndMaxRelativeTaps(pstRangeAction, linearProblem.infinity());
             double maxRelativeTap = Math.max(0, pstLimits.getRight());
-            double minRelativeTap = -Math.min(0, pstLimits.getLeft());
+            double minRelativeTap = Math.min(0, pstLimits.getLeft());
 
-            OpenRaoMPConstraint relativeTapConstraintUpward = linearProblem.addPstRelativeTapConstraint(0, maxRelativeTap, pstRangeAction, state, LinearProblem.VariationDirectionExtension.UPWARD);
             OpenRaoMPVariable preventivePstTapUpwardVariationVariable = linearProblem.getPstTapVariationVariable((PstRangeAction) preventiveRangeAction, optimizationPerimeter.getMainOptimizationState(), LinearProblem.VariationDirectionExtension.UPWARD);
-            relativeTapConstraintUpward.setCoefficient(pstTapUpwardVariationVariable, 1);
-            relativeTapConstraintUpward.setCoefficient(preventivePstTapUpwardVariationVariable, -1);
-
-            OpenRaoMPConstraint relativeTapConstraintDownard = linearProblem.addPstRelativeTapConstraint(0, minRelativeTap, pstRangeAction, state, LinearProblem.VariationDirectionExtension.DOWNWARD);
             OpenRaoMPVariable preventivePstTapDownwardVariationVariable = linearProblem.getPstTapVariationVariable((PstRangeAction) preventiveRangeAction, optimizationPerimeter.getMainOptimizationState(), LinearProblem.VariationDirectionExtension.DOWNWARD);
-            relativeTapConstraintDownard.setCoefficient(pstTapDownwardVariationVariable, 1);
-            relativeTapConstraintDownard.setCoefficient(preventivePstTapDownwardVariationVariable, -1);
+
+            OpenRaoMPConstraint relativeTapConstraint = linearProblem.addPstRelativeTapConstraint(minRelativeTap, maxRelativeTap, pstRangeAction, state);
```

Comment:
the bounds of this constraint need to take into consideration the "initial" tap of the psts in the fill method as well

Reply:
in the fill method (1st iteration), both taps are equal (= to the tap in the input network), aren't they? so they cancel each other out.
however I think we should take them into account in the update method